core/src/main/java/io/grpc/internal/GrpcUtil.java

```diff
@@ -70,11 +70,15 @@
 
   public static final Charset US_ASCII = Charset.forName("US-ASCII");
 
+  private static boolean isAppengineEnvironment() {
+    String appengineEnvironment = System.getProperty("com.google.appengine.runtime.environment");
```

Comment:
com.google.appengine.runtime.environment is either Production or Development. It has no other values.

The previous check was added explicitly. @louiscryan, would it hurt to remove the check?